annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddDockerConfigJsonSecretDecorator.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2018 The original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **/
+
+package io.dekorate.tekton.decorator;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+
+/**
+ * Special version of AddDockerConfigJsonSecretDecorator for Tekton where some tasks need the docker config file
+ * in .dockerconfigjson and some other tasks/steps require it in config.json.
+ *
+ * For example, the kaniko image explicitly requires the config.json file. See more in
+ * https://artifacthub.io/packages/tekton-task/tekton-tasks/kaniko?modal=manifest.
+ */
+public class AddDockerConfigJsonSecretDecorator extends io.dekorate.kubernetes.decorator.AddDockerConfigJsonSecretDecorator {
```

Comment:
We shouldn't need to extend this class just to use a different `file` name .

Reply:
Same comment than in https://github.com/dekorateio/dekorate/pull/1218#discussion_r1236705012